Clarifies scope broadening, fixing #30306 (#30308)
mdn/content: files/en-us/web/api/serviceworkercontainer/register/index.md

}
```

There is frequent confusion surrounding the meaning and use of _scope_.
A service worker can't have a scope broader than its own location,
unless the server specifies a broader maximum scope in a [Service-Worker-Allowed](https://w3c.github.io/ServiceWorker/#service-worker-allowed)
header on the service worker script. Therefore you should use the scope option

> when you need a narrower `scope` than the default.

Reviewer: this part may duplicate with the content on line 155-157

Author: I made a change to the paragraph on lines 155-157 to deal with the repetitive wording while trying to avoid a very abrupt change in flow after the previous code example.

}
```

As noted above, servers can change the default maximum scope by setting the `Service-Worker-Allowed` header on the service worker script. In this case, the `scope` option should specify a scope narrower than the header value, but potentially larger than the service worker's location.

> The following code, if included in `example.com/product/index.html`, would apply to all resources under `example.com` if the server set the `Service-Worker-Allowed` header to `/` or `https://example.com/`. If the server doesn't set the header, the service worker registration will fail, as the requested `scope` is too broad.

Reviewer 2: ```suggestion
The following code, if included in `example.com/product/index.html`, would apply to all resources under `example.com` if the server set the `Service-Worker-Allowed` header to `/` or `https://example.com/` when serving `sw.js`. If the server doesn't set the header, the service worker registration will fail, as the requested `scope` is too broad.
```

}
```

> As noted above, servers can remove the restriction on scope by setting the `Service-Worker-Allowed` header on the service worker script, specifying a maximum scope for that service worker above the service worker's location.

Reviewer 2: ```suggestion
As noted above, servers can change the default maximum scope by setting the `Service-Worker-Allowed` header on the service worker script. In this case, the `scope` option should specify a scope narrower than the header value, but potentially larger than the service worker's location.
```

worker script, and then you can specify a max scope for that service worker above the
service worker's location.

> The following code, if included in `example.com/index.html`, would apply to all
> resources under `example.com` if the server set the Service-Worker-Allowed header
> to `/` or `https://example.com/`. If the server doesn't set the header,
> the service worker registration will fail, as the requested `scope` is too broad.

Reviewer: ```suggestion
The following code, if included in `example.com/index.html`, would apply to all
resources under `example.com` if the server set the `Service-Worker-Allowed` header
to `/` or `https://example.com/`. If the server doesn't set the header,
the service worker registration will fail, as the requested `scope` is too broad.
```

> The following code, if included in `example.com/index.html`, would apply to all
> resources under `example.com` if the server set the Service-Worker-Allowed header
> to `/` or `https://example.com/`. If the server doesn't set the header,
> the service worker registration will fail, as the requested `scope` is too broad.

Reviewer: ```suggestion
The following code, if included in `example.com/index.html`, would apply to all
resources under `example.com` if the server set the `Service-Worker-Allowed` header
to `/` or `https://example.com/`. If the server doesn't set the header,
the service worker registration will fail, as the requested `scope` is too broad.
```
